### DISCUSSION TITLE
Installing Drupal 9 in non-default folder

### DISCUSSION THREAD

**Discussion by dcesystems** [ddev/ddev#3846] [Q&A] [answered]
Hello,
when I try to install drupal in a webroot folder other than the default "web" I am having problems.

According to the  drupal documentation, I need to run

composer create-project --no-install drupal/recommended-project my_project

Edit composer.json with the desired docroot folder name. Composer install is then executed from the my_project directory. The files created in the my_project is then copied into the directory containing the server document root directory.

I get an error when I try to run ...composer create-project --no-install drupal/recommended-project my_project

the error is:

"Creating a "drupal/recommended-project" project at "./my_project"

In VersionParser.php line 521:
                                                                                        
  Could not parse version constraint /tmp/MUhdIg: Invalid version string "/tmp/MUhdIg" "

So, is there a way I can in install drupal 9 in a  directory other than the default "web" directory? 

Thanks in advance

**Comment by rfay** [comment, marked as answer]
This doesn't really have anything to do with ddev, it has to do with the drupal/recommended project. 

Please experiment with the Drupal 9 quickstart, https://ddev.readthedocs.io/en/latest/users/cli-usage/#drupal-9-quickstart 

If you really have to change the "web" directory to something else, you'll have to
1. Change the initial setup to `ddev config --project-type=drupal9 --docroot=<something> --create-docroot
`
2. `ddev composer create "drupal/recommended-project" --no-install` and then edit composer.json to change "web/" to something else everywhere it appears, for example, change `web/` to `docroot/`
3. `ddev composer install`

But the reality is... you don't want to use drupal/recommended if you don't want to use "web" as the docroot.

**Reply by dcesystems** [reply]
I had tried the procedures you mentioned (ddev config, ddev composer create etc). The example I gave in the parent post was me experimenting afterwards to to try and get things to work (trying composer create-project command, passing project directory parameters etc as per drupal documentation). I should have elaborated more.

I got it to work by deleting the "other-dir" (using it as the example folder) right after running ddev config --project-type=drupal9 --docroot=other-dir --create-docroot

Then ddev start

ddev composer create "drupal/recommended-project" --no-install

edit composer.json

ddev composer install 

and then ddev launch

For some reason, this got things going.

I am glad I got this to work. Quick and easy to spin up local instances and working local ssl certs out of the box. Good stuff.

Thanks for taking the time to assist me, much appreciated.. cheers!

**Reply by rfay** [reply]
Glad it's working for you! For most people, sticking with the defaults is a good idea :) Avoids things like this.
